src/hevm/src/EVM/SymExec.hs

```diff
@@ -45,33 +45,31 @@ sbytes1024 = liftA2 (++) sbytes512 sbytes512
 -- We don't assume input types are restricted to their proper range here;
 -- such assumptions should instead be given as preconditions.
 -- This could catch some interesting calldata mismanagement errors.
-symAbiArg :: AbiType -> Query ([SWord 8], SWord 32)
-symAbiArg (AbiUIntType n) | n `mod` 8 == 0 && n <= 256 = do x <- sbytes32
-                                                            return (x, 32)
-                          | otherwise = error "bad type"
+symAbiArg :: AbiType -> Query [SWord 8]
+symAbiArg (AbiUIntType n)
+  | n `mod` 8 == 0 && n <= 256 = sbytes32
+  | otherwise = error "bad type"
 
-symAbiArg (AbiIntType n)  | n `mod` 8 == 0 && n <= 256 = do x <- sbytes32
-                                                            return (x, 32)
-                          | otherwise = error "bad type"
-symAbiArg AbiBoolType = do x <- sbytes32
-                           return (x, 32)
+symAbiArg (AbiIntType n)
+  | n `mod` 8 == 0 && n <= 256 = sbytes32
+  | otherwise = error "bad type"
 
-symAbiArg AbiAddressType = do x <- sbytes32
-                              return (x, 32)
+symAbiArg AbiBoolType = sbytes32
 
-symAbiArg (AbiBytesType n) | n <= 32 = do x <- sbytes32
-                                          return (x, 32)
-                           | otherwise = error "bad type"
+symAbiArg AbiAddressType = sbytes32
+
+symAbiArg (AbiBytesType n)
+  | n <= 32 = sbytes32
+  | otherwise = error "bad type"
 
 -- TODO: is this encoding correct?
```

Comment:
Is this `TODO` still valid?

Reply:
It hasn't really seen any testing yet so I'm not sure